src/cascadia/ShellExtension/OpenTerminalHere.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "OpenTerminalHere.h"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+
+static std::wstring VerbDisplayName{ L"Open in Windows Terminal" };
+static std::wstring VerbDevBuildDisplayName{ L"Open in Windows Terminal (Dev Build)" };
+static std::wstring VerbName{ L"WindowsTerminalOpenHere" };
+
+// This code is aggressively copied from
+//   https://github.com/microsoft/Windows-classic-samples/blob/master/Samples/
+//   Win7Samples/winui/shell/appshellintegration/ExplorerCommandVerb/ExplorerCommandVerb.cpp
+
+// Function Description:
+// - This is a helper to determine if we're running as a part of the Dev Build
+//   Package or the release package. We'll need to return different text, icons,
+//   and use different commandlines depending on which one the user requested.
+// - Uses a C++11 "magic static" to make sure this is only computed once.
+// - If we can't determine if it's the dev build or not, we'll default to true
+// Arguments:
+// - <none>
+// Return Value:
+// - true if we believe this extension is being run in the dev build package.
+bool IsDevBuild()
+{
+    // use C++11 magic statics to make sure we only do this once.
+    static bool isDevBuild = []() -> bool {
+        try
+        {
+            const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
+            const auto id = package.Id();
+            const std::wstring name{ id.FullName() };
+            // Does our PFN start with WindowsTerminalDev?
+            return name.rfind(L"WindowsTerminalDev", 0) == 0;
+        }
+        CATCH_LOG();

```

Comment:
we might need a more robust handler to get the exe name like the one in winconpty.cpp... if we do eventually deliver an MSI, i'd prfer the shell ext to do a sxs lookup to find WindowsTerminal.exe next to it.

Reply:
That's... probably right tbh